Index: master/src/backend/bacon-video-widget-gst-missing-plugins.c
===================================================================
--- master.orig/src/backend/bacon-video-widget-gst-missing-plugins.c
+++ master/src/backend/bacon-video-widget-gst-missing-plugins.c
@@ -233,72 +233,79 @@ bacon_video_widget_start_plugin_installa
 
 	return TRUE;
 }
+void show_codec_confirmation_dialog(TotemCodecInstallContext *ctx, const gchar *install_helper_display_name);
 
-static void
-codec_confirmation_dialog_response_cb (GtkDialog       *dialog,
-                                       GtkResponseType  response_type,
-                                       gpointer         user_data)
-{
-	TotemCodecInstallContext *ctx = user_data;
-
-	switch (response_type) {
-	case GTK_RESPONSE_ACCEPT:
-		bacon_video_widget_start_plugin_installation (ctx, FALSE);
-		break;
-	case GTK_RESPONSE_CANCEL:
-	case GTK_RESPONSE_DELETE_EVENT:
-		break;
-	default:
-		g_assert_not_reached ();
-	}
-	gtk_widget_destroy (GTK_WIDGET (dialog));
+// Função para encerrar e reiniciar a aplicação Totem
+static void restart_totem_application() {
+    // Comando para encerrar o Totem
+    g_spawn_command_line_async("pkill totem", NULL);
+
+    // Adiciona um atraso para garantir que o Totem seja encerrado antes de reiniciá-l
+    
+	gchar *cmd = "env GDK_BACKEND=x11 /usr/share/policorp-codecs/codec-multimedia";
+    GError *error = NULL;
+    gint exit_status;
+
+    if (!g_spawn_command_line_sync(cmd, NULL, NULL, &exit_status, &error)) {
+        g_printerr("Failed to execute command: %s\n", error->message);
+        g_clear_error(&error);
+    }
+
+    cmd = "totem"; 
+    if (!g_spawn_command_line_async(cmd, NULL)) {
+        g_printerr("Failed to start Totem\n");
+    }
 }
 
-static void
-show_codec_confirmation_dialog (TotemCodecInstallContext *ctx,
-                                const gchar              *install_helper_display_name)
-{
-	GtkWidget *button;
-	GtkWidget *dialog;
-	GtkWidget *toplevel;
-	gchar *button_text;
-	gchar *descriptions_text;
-	gchar *message_text;
-
-	toplevel = gtk_widget_get_toplevel (GTK_WIDGET (ctx->bvw));
-
-	dialog = gtk_message_dialog_new (GTK_WINDOW (toplevel),
-	                                 GTK_DIALOG_MODAL |
-	                                 GTK_DIALOG_DESTROY_WITH_PARENT,
-	                                 GTK_MESSAGE_ERROR,
-	                                 GTK_BUTTONS_CANCEL,
-	                                 _("Unable to play the file"));
-
-	descriptions_text = g_strjoinv (", ", ctx->descriptions);
-	message_text = g_strdup_printf (ngettext ("%s is required to play the file, but is not installed.",
-	                                          "%s are required to play the file, but are not installed.",
-	                                          g_strv_length (ctx->descriptions)),
-	                                descriptions_text);
-
-	/* TRANSLATORS: this is a button to launch a codec installer.
-	 * %s will be replaced with the software installer's name, e.g.
-	 * 'Software' in case of gnome-software. */
-	button_text = g_strdup_printf (_("_Find in %s"), install_helper_display_name);
-	gtk_message_dialog_format_secondary_text (GTK_MESSAGE_DIALOG (dialog), "%s", message_text);
-	button = gtk_dialog_add_button (GTK_DIALOG (dialog),
-	                                button_text,
-	                                GTK_RESPONSE_ACCEPT);
-	gtk_dialog_set_default_response (GTK_DIALOG (dialog), GTK_RESPONSE_ACCEPT);
-	gtk_style_context_add_class (gtk_widget_get_style_context (button), "suggested-action");
-	g_signal_connect (dialog, "response",
-	                  G_CALLBACK (codec_confirmation_dialog_response_cb),
-	                  ctx);
-
-	gtk_window_present (GTK_WINDOW (dialog));
-
-	g_free (button_text);
-	g_free (descriptions_text);
-	g_free (message_text);
+static void 
+codec_confirmation_dialog_response_cb(GtkDialog *dialog, gint response_id, gpointer user_data) {
+
+	if (response_id == GTK_RESPONSE_ACCEPT) {
+        	restart_totem_application();
+   	 }
+
+    	gtk_widget_destroy(GTK_WIDGET(dialog));
+}
+
+void show_codec_confirmation_dialog(TotemCodecInstallContext *ctx, const gchar *install_helper_display_name) {
+    GtkWidget *button;
+    GtkWidget *dialog;
+    GtkWidget *toplevel;
+    gchar *button_text;
+    gchar *descriptions_text;
+    gchar *message_text;
+
+    toplevel = gtk_widget_get_toplevel(GTK_WIDGET(ctx->bvw));
+
+    dialog = gtk_message_dialog_new(GTK_WINDOW(toplevel),
+                                    GTK_DIALOG_MODAL |
+                                    GTK_DIALOG_DESTROY_WITH_PARENT,
+                                    GTK_MESSAGE_ERROR,
+                                    GTK_BUTTONS_CANCEL,
+                                    _("Unable to play the file"));
+
+    descriptions_text = g_strjoinv(", ", ctx->descriptions);
+    message_text = g_strdup_printf(ngettext("%s is required to play the file, but is not installed.",
+                                            "%s are required to play the file, but are not installed.",
+                                            g_strv_length(ctx->descriptions)),
+                                   descriptions_text);
+
+    button_text = g_strdup_printf(_("Abrir instalador de codecs"));
+    gtk_message_dialog_format_secondary_text(GTK_MESSAGE_DIALOG(dialog), "%s", message_text);
+    button = gtk_dialog_add_button(GTK_DIALOG(dialog),
+                                   button_text,
+                                   GTK_RESPONSE_ACCEPT);
+    gtk_dialog_set_default_response(GTK_DIALOG(dialog), GTK_RESPONSE_ACCEPT);
+    gtk_style_context_add_class(gtk_widget_get_style_context(button), "suggested-action");
+    g_signal_connect(dialog, "response",
+                     G_CALLBACK(codec_confirmation_dialog_response_cb),
+                     NULL);
+
+    gtk_window_present(GTK_WINDOW(dialog));
+
+    g_free(button_text);
+    g_free(descriptions_text);
+    g_free(message_text);
 }
 
 static void
